### PR TITLE
Toggle file switcher when pressing its keyboard shortcut

### LINKED ISSUE #3297: Pressing Ctrl+P twice opens file switcher twice
## Description of your problem

**What steps will reproduce the problem?**

1. Press Ctrl-P
2. Press Ctrl-P again

**What is the expected output? What do you see instead?**

The first Ctrl-P opens the selection pop up. The second Ctrl-P should (preferably) close it while open, or at least do nothing. Instead, it seems a second pop up is opened over the first one (you then have to press Esc twice to close both of them).

## Versions and main components

* Spyder Version: Spyder 3.0.0.dev0 (from `spyder-ide` conda channel)
* Python Version: 3.5.2
* Operating system: Linux

## Dependencies

Please go to the menu entry `Help > Optional Dependencies` (or
`Help > Dependencies`), press the button `Copy to clipboard`
and paste the contents below:

jedi >=0.8.1     :  0.9.0 (OK)
matplotlib >=1.0 :  None (NOK)
nbconvert >=4.0  :  4.2.0 (OK)
numpy >=1.7      :  None (NOK)
pandas >=0.13.1  :  None (NOK)
pep8 >=0.6       :  1.7.0 (OK)
psutil >=0.3     :  4.3.0 (OK)
pyflakes >=0.6.0 :  1.2.3 (OK)
pygments >=2.0   :  2.1.3 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.4.1 (OK)
sympy >=0.7.3    :  None (NOK)

### PR DESCRIPTION
**Pull request by ccordoba12**
Fixes #3297

### CODE REVIEW

File: spyderlib/widgets/editor.py [hunk 1]
```diff
@@ -581,10 +581,16 @@ def open_fileswitcher_dlg(self):
         """Open file list management dialog box"""
         if not self.tabs.count():
             return
+        if self.fileswitcher_dlg is not None and \
+          self.fileswitcher_dlg.is_visible:
+            self.fileswitcher_dlg.hide()
+            self.fileswitcher_dlg.is_visible = False
```
Comment (Nodd) [inline spyderlib/widgets/editor.py]: This is already supported by Qt, with the `visible` property. See http://doc.qt.io/qt-4.8/qwidget.html#visible-prop

Comment (ccordoba12) [inline spyderlib/widgets/editor.py]: Ops, I didn't know about it. I looked for `isVisible` but it seems QDialogs don't support it.

I'll change it in master.

Comment (ccordoba12) [inline spyderlib/widgets/editor.py]: It doesn't with Qt5, so it was probably only available with Qt4.

Comment (ccordoba12) [inline spyderlib/widgets/editor.py]: So I'll leave things as they are ;-)

Comment (Nodd) [inline spyderlib/widgets/editor.py]: It is still present in Qt5: http://doc.qt.io/qt-5.6/qwidget.html#visible-prop (and QDialog inherits QWidget). So it should work...
Anyway it's not a big deal, this comment is for completeness only.

Comment (ccordoba12) [inline spyderlib/widgets/editor.py]: I tried to use `visible`, believe me ;-) It just didn't work, I don't know why...

Comment (Nodd) [inline spyderlib/widgets/editor.py]: I think I found why. The widget is created on demand, and during initialization the widget is not visible yet so a check with `visible()` will be wrong. Your code is as good as any other workaround.

Comment (ccordoba12) [inline spyderlib/widgets/editor.py]: Ok, good to know ;-)

**Pull request merged**